packages/kit/src/core/load_config/index.js

```diff
@@ -81,7 +82,10 @@ function remove_trailing_slash(str) {
 }
 
 export async function load_config({ cwd = process.cwd() } = {}) {
-	const config_file = path.join(cwd, 'svelte.config.cjs');
+	const config_file_esm = path.join(cwd, 'svelte.config.js');
+	const config_file = fs.existsSync(config_file_esm)
+		? config_file_esm
+		: path.join(cwd, 'svelte.config.cjs');
 	const config = await import(url.pathToFileURL(config_file).href);
 	const validated = validate_config(config.default);
```

Comment:
I think an extra await here would be all that would be needed if we wanted to support files that exported a promise resolving to the configuration.